ls1intum/Artemis
# [Feature] Remove Instruction slide toggle and combine exercise instruction text with structured assessment instructions in exercise create/update

### Checklist

### Motivation and Context



PR covers that removing the slide toggle element in exercise creation/update page and using single markdown editor for grading instructions if the assessment period has not been started. After assessment period, each structured grading instructions have separate editor, now with this PR, grading instructions of exercise also has separate editor in form.

Now we can change/re-evaluate grading instructions after assessment period, we do not need to separate the instructions text with structured grading instructions. We can use single markdown editor during the exercise creation.  Therefore, we do not need slide toggle element.

### Description


- Slide toggle component removed from Artemis shared modules (since it is used only for switching grading instructions in exercise creation page)
- Removed slide toggle component and  markdown editor for grading instructions from create/update page of each type of exercises
- If the assessment has not been started, there is one editor named Assessment Instructions. The text before [criterion] or [instruction] tag is assigned as `gradingInstructions` of `exercise`. 
- added client tests

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Management 
3. Create new or update existing exercise (could be anyone except quiz)
4. 
- If you did not use structured grading instructions during assessment for the exercise you created or updated:
a. you should see single editor for Assessment Instructions
b. There should be "Add Assessment Instruction text here" initial text before [instruction] or [criterion] tag. Keep or modify this text, then save the changes for exercise. 
c. In the exercise detail page, you should see "Assessment Instructions" and Structured Assessment Instructions".
d. Texts must be assigned properly.
e. Go edit page again, remove the text before [instruction] or [criterion] tag. Then in the exercise detail page, Assessment Instructions won't be there.
- If you  used structured grading instructions during assessment for the exercise you created or updated:
a. You should see Assessment Instructions in a form view.
b. The first box is used for `gradingInstructions` of `exercise`. So, add some text there or modify the existing text. Also make some changes for structured grading instructions. Save changes for exercise.
c. In the exercise detail page, you should see "Assessment Instructions" and Structured Assessment Instructions".
d. Texts must be assigned properly.

### Review Progress





- Code Review
- Manual Tests

### Test Coverage




grading-instructions-details.component.ts 57% (will be follow-ups, I can increase the coverage later on)

### Screenshots



Exercise Edit Page - Grading instructions are not used in the assessment

Before|After
------|------
![grafik](https://user-images.githubusercontent.com/23650630/129626134-687d7764-8b37-4834-9f95-ebc265d5daed.png)|![grafik](https://user-images.githubusercontent.com/23650630/129626069-07e9de2f-0010-4105-b58e-c33cc660cbf9.png)

Exercise Edit Page - Grading instructions are used in the assessment

Before|After
------|------
![grafik](https://user-images.githubusercontent.com/23650630/129626269-531038da-300c-448b-8c60-971e899bb4a5.png)|![grafik](https://user-images.githubusercontent.com/23650630/129625897-93cce254-be06-4c24-bbe9-db85fdc60f3e.png)
